amy/emails/tests/actions/test_instructor_signs_up_for_workshop.py

```diff
@@ -0,0 +1,266 @@
+from datetime import UTC, date, datetime, timedelta
+from unittest import mock
+import weakref
+
+from django.test import RequestFactory, TestCase, override_settings
+from django.urls import reverse
+
+from communityroles.models import CommunityRole, CommunityRoleConfig
+from emails.actions import instructor_signs_up_for_workshop_receiver
+from emails.models import EmailTemplate, ScheduledEmail
+from emails.signals import instructor_signs_up_for_workshop_signal
+from recruitment.models import InstructorRecruitment, InstructorRecruitmentSignup
+from workshops.models import Event, Organization, Person
+from workshops.tests.base import TestBase, consent_to_all_required_consents
+
+
+class TestInstructorSignsUpForWorkshopReceiver(TestCase):
+    @mock.patch("emails.utils.logger")
+    def test_disabled_when_no_feature_flag(self, mock_logger) -> None:
+        # Arrange
+        with self.settings(EMAIL_MODULE_ENABLED=False):
+            # Act
+            instructor_signs_up_for_workshop_receiver(None)
+            # Assert
+            mock_logger.debug.assert_called_once_with(
+                "EMAIL_MODULE_ENABLED not set, skipping receiver "
+                "instructor_signs_up_for_workshop_receiver"
+            )
+
+    def test_signal_received(self) -> None:
+        # Arrange
+        organization = Organization.objects.first()
+        event = Event.objects.create(
+            slug="test-event", host=organization, administrator=organization
+        )
+        recruitment = InstructorRecruitment.objects.create(
+            event=event, notes="Test notes"
+        )
+        person = Person.objects.create()
+        signup = InstructorRecruitmentSignup.objects.create(
+            recruitment=recruitment, person=person
+        )
+        request = RequestFactory().get("/")
+        mock_action = mock.MagicMock()
+        _copied_receivers = instructor_signs_up_for_workshop_signal.receivers[:]
+
+        # This hack replaces weakref to "emails.actions.persons_merged_receiver" with
```

Comment:
```suggestion
        # This hack replaces weakref to "emails.actions.instructor_signs_up_for_workshop_receiver" with
```

Reply:
I decided to drop this test since it wasn't really testing that "our" receiver received a signal (this we should trust Django to do properly). On top of that we were using "hacks" to substitute a mocked weakref, so we were not really testing "our" receiver at all.

The tests like this one were replaced with simpler "make sure that specific receiver is connected to the signal".